components/InputStates.js

```diff
@@ -0,0 +1,64 @@
+/**
+ * InputStates.js
+ * 
+ * Manages the input states for a component's ports
+ */
+
+var _ = require('underscore');
+
+var inputStates = [];
```

Comment:
JS noobie question here.  Does this create a singleton instance -- one per InputStates.js module?  It looks like it does.  But if so, then I don't understand how we will use this module to create one instance per VNI instance.  I've been searching around and have not been able to find an answer.  Maybe you can give me a quick explanation when you can.